app/controllers/concerns/foreman/controller/parameters/discovered_host.rb

```diff
@@ -0,0 +1,24 @@
+module Foreman::Controller::Parameters::DiscoveredHost
+  extend ActiveSupport::Concern
+  include Foreman::Controller::Parameters::HostBase
+  include Foreman::Controller::Parameters::HostCommon
+
+  class_methods do
+    def discovered_host_params_filter
+      Foreman::ParameterFilter.new(::Host::Discovered, :top_level_hash => 'host').tap do |filter|
+        filter.permit :discovery_rule_id
+
+        add_host_base_params_filter(filter)
+        add_host_common_params_filter(filter)
+      end
+    end
+  end
+
+  def discovered_host_params
+    self.class.discovered_host_params_filter.filter_params(params, parameter_filter_context)
+  end
+
+  def discovered_host_search_params
+    params.permit(:search, :order, :page)
+  end
```

Comment:
These are not mass-assigned, I will remove the helper here and in `discovery_rule` as well.